src/components/WorkBuddies/MyWorkBuddy.tsx

```diff
@@ -0,0 +1,118 @@
+import { Fragment, useContext, useEffect, useState } from "react";
+import { BuddiesContext } from "../BuddiesContext";
+import classNames from "classnames";
+import "./work-buddies.css";
+import { useReducer } from "react";
+import { Buddy } from "./BuddyInterface";
+import { memo } from "react";
+
+interface propsInterface {
+  myBuddy: Buddy;
+}
+
+const reducer = (
+  index: number,
+  action: { type: string; contextData?: Buddy[]; myBuddyIndex?: number }
+) => {
+  switch (action.type) {
+    case "next":
+      if (action.contextData) {
+        if (index + 1 < action.contextData.length) return index + 1;
+        else return 0;
+      } else return 0;
+
+    case "prev":
+      if (action.contextData) {
+        if (index - 1 > -1) return index - 1;
+        else return action.contextData.length - 1;
+      } else return 0;
+
+    case "my-buddy":
+      if (action.myBuddyIndex) {
+        return action.myBuddyIndex;
+      } else return 0;
+
+    default:
+      return 1;
+  }
+};
+
+const MyWorkBuddy = memo(({ myBuddy }: propsInterface) => {
+  const { contextData } = useContext(BuddiesContext);
+
+  //useReducer
+  const [currentBuddyIndex, dispatch] = useReducer(reducer, 0);
+  const [buddy, setBuddy] = useState<Buddy>();
+  const [myBuddyIndex, setMyBuddyIndex] = useState<number>();
+
+  //I am aware that I can get "buddy" from contextData and that I'm not actually using it,
```

Comment:
Is this related to my question on line **45**?

Reply:
Yes